CougSat1-ADCS/src/drivers/ADCS.cpp

```diff
@@ -16,38 +16,63 @@
  */
 
 #include "ADCS.h"
+#define CDHFLAG 0x1
+#define ONEMILLISECOND 1
+#define ONESECOND 1000
 
-void ADCS::monitor_thread() {
-    while (true) {
-        Thread::wait(1);
-        // printf("Checking for events!\r\n");
-        if(Reader.checkRead()) {
-            I2CREAD.signal_set(0x1);
+/**
+ * @brief Function for the polling thread
+ * 
+ */
+void ADCS::monitor_thread()
+{
+    while (true)
+    {
+        Thread::wait(ONEMILLISECOND);
```

Comment:
Are you really going to multi-thread with all of the associated troubles? If not, use the mbed wait: wait_ms()

Reply:
I don't think your subsystem will benefit from multi-threading. You can, and should, check that there is a message and read it pretty immediately. Acting on it can be later but as is, the threads look like unnecessary complications.

Reply:
The thread is the name of the priority queue in mbed. It has multiple threads but they are all executing sequentially. This allows us to prioritize tasks. This does not do multithreading. This only pauses the task long enough to allow other threads that are waiting to execute. Then one milisecond later will place this thread back in the queue